src/profile-logic/process-profile.js

```diff
@@ -103,11 +103,29 @@ function _cleanFunctionName(functionName: string): string {
   return functionName;
 }
 
-function _extractFuncsAndResourcesFromFrames(
-  geckoFrameStruct: GeckoFrameStruct,
+type ExtractionInfo = {
+  funcTable: FuncTable,
+  resourceTable: ResourceTable,
+  stringTable: UniqueStringArray,
+  libs: Lib[],
+  libToResourceIndex: Map<Lib, IndexIntoResourceTable>,
+  originToResourceIndex: Map<string, IndexIntoResourceTable>,
+  libNameToResourceIndex: Map<IndexIntoStringTable, IndexIntoResourceTable>,
+  stringTableIndexToNewFuncIndex: Map<IndexIntoStringTable, IndexIntoFuncTable>,
+};
+
+/**
+ * Resources and funcs are not part of the Gecko Profile format. This information is
+ * implicitly defined in the frame tables' location strings. This function derives a new
+ * FuncTable and ResourceTable for easily accesing this information in a structred format.
```

Comment:
Please also mention that the `IndexIntoFuncTable[]` return value maps the index of each element in the locationStringIndexes array to a func from the returned FuncTable.